x-pack/plugins/ml/server/saved_objects/repair.ts

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from '@hapi/boom';
+import { IScopedClusterClient } from 'kibana/server';
+import type { JobObject, JobSavedObjectService } from './service';
+import { JobType } from '../../common/types/saved_objects';
+import { checksFactory } from './checks';
+
+import { Datafeed } from '../../common/types/anomaly_detection_jobs';
+
+export function repairFactory(
+  client: IScopedClusterClient,
+  jobSavedObjectService: JobSavedObjectService
+) {
+  const { checkStatus } = checksFactory(client, jobSavedObjectService);
+
+  async function repairJobs(simulate: boolean = false) {
+    type Result = Record<string, { success: boolean; error?: any }>;
+    const results: {
+      savedObjectsCreated: Result;
+      savedObjectsDeleted: Result;
+      datafeedsAdded: Result;
+      datafeedsRemoved: Result;
+    } = {
+      savedObjectsCreated: {},
+      savedObjectsDeleted: {},
+      datafeedsAdded: {},
+      datafeedsRemoved: {},
+    };
+
+    const { body: datafeeds } = await client.asInternalUser.ml.getDatafeeds<{
+      datafeeds: Datafeed[];
+    }>();
+
+    const tasks: Array<() => Promise<void>> = [];
+
+    const status = await checkStatus();
+    for (const job of status.jobs['anomaly-detector']) {
+      if (job.checks.savedObjectExits === false) {
+        if (simulate === true) {
+          results.savedObjectsCreated[job.jobId] = { success: true };
+        } else {
+          // create AD saved objects for jobs which are missing them
+          const jobId = job.jobId;
+          const datafeedId = job.datafeedId;
+          tasks.push(async () => {
+            try {
+              await jobSavedObjectService.createAnomalyDetectionJob(jobId, datafeedId ?? undefined);
+              results.savedObjectsCreated[job.jobId] = { success: true };
+            } catch (error) {
+              results.savedObjectsCreated[job.jobId] = {
+                success: false,
+                error: error.body ?? error,
+              };
+            }
+          });
+        }
+      }
```

Comment:
nit: Avoid extra indentation and make readability a bit better. Applicable for the `for...of` statement below as well 
```suggestion
if (job.checks.savedObjectExits) return;

if (simulate === true) {
    results.savedObjectsCreated[job.jobId] = {success: true};
    return;
}

// create AD saved objects for jobs which are missing them
const jobId = job.jobId;
const datafeedId = job.datafeedId;
tasks.push(async () => {
    try {
        await jobSavedObjectService.createAnomalyDetectionJob(jobId, datafeedId ?? undefined);
        results.savedObjectsCreated[job.jobId] = {success: true};
    } catch (error) {
        results.savedObjectsCreated[job.jobId] = {
            success: false,
            error: error.body ?? error,
        };
    }
});
```

Reply:
`return` breaks the loop, these would need to be `continue`

Reply:
i agree that these changes would improve the readability of this code, however this function is not new and until we have tests to cover it i'd like to avoid refactoring that could introduce bugs.